app/assets/javascripts/dashboard.coffee

```diff
@@ -25,6 +26,24 @@ synchronizeCourse = (event) ->
       $("div.user-courses-container").html(data.partial)
   event.preventDefault()
 
+synchronizeDatesDashboard = (event) ->
+  url = "/user_dates/synchronize_dates_on_dashboard.json"
+  $.ajax
+    url: url
+    method: 'GET'
+    error: (jqXHR, textStatus, errorThrown) ->
+      console.log('error_synchronize_dates')
+      alert(I18n.t('global.ajax_failed'))
+    success: (data, textStatus, jqXHR) ->
+      console.log('success_synchronize_dates')
+      for mooc_provider, result of data.synchronization_state
+        if result == false
+          alert(I18n.t('dashboard.error_synchronize_dates', provider: mooc_provider))
+        else if result != true
+          window.location.replace(result)
```

Comment:
Do you need to redirect the user when synchronizing dates? I know that this line is similar to the line regarding the courses. Above, it is needed due to the 30 min session time out at Coursera with their current OAuth implementation.